controllers/clustergroupupgrade_controller.go

```diff
@@ -446,6 +443,33 @@ func (r *ClusterGroupUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.
 				return
 			}
 
+			if clusterGroupUpgrade.GetAnnotations()[utils.BlockingCRCompletionModeAnn] == utils.BlockingCRCompletionModePartial {
+				clusters, err = r.filterNonCompletedClustersInBlockingCRs(ctx, clusterGroupUpgrade, clusters)
+				if err != nil {
+					r.Log.Error(err, "filterNonCompletedClustersInBlockingCRs")
+					return
+				}
+				if len(clusters) == 0 {
+					utils.SetStatusCondition(
+						&clusterGroupUpgrade.Status.Conditions,
+						utils.ConditionTypes.Progressing,
+						utils.ConditionReasons.Completed,
+						metav1.ConditionFalse,
+						"No clusters available for remediation after filtering non completed clusters",
```

Comment:
Maybe more clear with this?
```suggestion
"No clusters available for remediation after filtering out non-completed clusters from blocking CGUs",
```